src/store/toastStore/ToastsStore.ts

```diff
@@ -0,0 +1,27 @@
+import { makeAutoObservable } from 'mobx';
+import { IToast, SeverityEnum } from './ToastsStore.type';
+
+class ToastsStoreClass {
+  freeId: number;
+  toasts$: Array<IToast | undefined>;
+  delay: number;
+
+  constructor(delay: number) {
+    this.delay = delay;
+    this.freeId = 0;
+    this.toasts$ = [];
+    makeAutoObservable(this);
```

Comment:
makeObservable(this, {
   toasts$: observable,
    addToast: action,
})

action = () => 
action.bound () {}